portfolio/src/main/webapp/script.js

```diff
@@ -30,32 +30,57 @@ function addRandomFact() {
 /**
  * Controls slideshow image display on the page
  */
-var slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
-for(var i = 0; i < slideIndices.length; i++) {
-    showSlides(i, slideIndices[i]);
+const decks = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const SLIDES = [["/images/dog/dog0.jpg", "/images/dog/dog1.jpg", "/images/dog/dog2.jpg"],
+  ["/images/place/place0.jpg", "/images/place/place1.jpg", "/images/place/place2.jpg"],
+  ["/images/food/food0.jpg", "/images/food/food1.jpg", "/images/food/food2.jpg"]]
+
+let slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
+
+function changeSlide(deckNum, direction) {
+  const deckSize = SLIDES[deckNum].length;
+  let slideIndex = slideIndices[deckNum];
+  slideIndex += direction;
+  if(slideIndex === deckSize) { // wrap around
+      slideIndex = 0;
+  }
+  else if(slideIndex === -1) {
+      slideIndex = deckSize - 1;
+  }
+  slideIndices[deckNum] = slideIndex;
 }
 
-function nextSlide(deckNum, direction) {
-  showSlides(deckNum, slideIndices[deckNum] += direction);
+document.getElementById("prev-dog-slide-button").onclick = function() {
+  changeSlide(decks.DOG_DECK, -1);
```

Comment:
Just by reading this line, it is not cleared what the -1 does. It can be ambiguous. I can guess what deck.DOG_DECK is used for.

Maybe change have on top of this file something like`const PREV_SLIDE_DIR = -1` and `const NEXT_SLIDE_DIR = 1` and then use that constant variable here.